src/inkypi.py

```
@@ -574,6 +575,26 @@ def _set_security_headers(response):
             pass
         return response
 
+    # Register signal handlers for graceful shutdown (main process only)
+    if not is_running_from_reloader():
+
+        def _shutdown_handler(signum, frame):
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `_shutdown_handler`

Add return type annotation: `NoReturn`

(ANN202)